WordPress/openverse-frontend
# Use JSON5 for `en.json`

## Fixes


Fixes #1815 by @dhruvkb

## Description


This PR 
- replaces `en.json` with `en.json5` and reformats it
- adds a rudimentary JSON5 parser for the `en.json5` file
  - could not find any parser that's capable of retaining comments when the JSON5 file is parsed so made a basic one specifically for reading i18n files 
- updates PO generation functionality to include comments
  - updates POT file to include all references to the i18n string 
  - updates POT file to preserve the order of the `en.json5` file
  - uses the right type of POT comment (`#.` vs `#:`)
- updates scripts that used the `en.json` file
- converts English `en.json` to be like any other language
  - added JSON5 &rarr; JSON conversion for English to `i18n:get-translations` script
- makes general refactors, shortens code, adds docstrings with types, and cleans up code

Code in many places depends on `en.json`. This continues to work once the `en.json` file has been generated by running `i18n:get-translations`.

## Screenshots

| en.json5 | openverse.pot |
|-|-|
| <img width="504" alt="Screenshot 2022-10-30 at 6 11 15 PM" src="https://user-images.githubusercontent.com/16580576/198883191-2ddd1a63-f740-479b-b52f-f523d684cf06.png"> | <img width="484" alt="Screenshot 2022-10-30 at 6 11 47 PM" src="https://user-images.githubusercontent.com/16580576/198883214-da5e232f-0587-4b06-a59f-c9145d360688.png"> |

## Testing Instructions

0. Check out the PR.
1. Setup i18n `pnpm i18n`.
2. Run the dev server and see that everything is translated as expected.

Additionally CI is passing and unit tests that depend on locales are passing. So it's good to start. 


## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
